### PR TITLE
Urgent fix to Recipe-Ingredient relationship

### PR DESCRIPTION
**Pull request by Dissurender**
Summary: recipe-ingredient relationship was making a self-referencing loop.

### Fixes to mitigate this happening
- Changing Recipe-Ingredient to One-to-Many unidirectional
- Remove logic that could cause infinite recursion
- moved repository saves to be +1 scope from the appropriate methods for readability

### CODE REVIEW

**Review by bduran04 — APPROVED** [review]
roger, roger

File: Backend_Application/src/main/java/dsd/cohort/application/bootstrap/BootstrapData.java
Comment (wcouturiaux) [inline Backend_Application/src/main/java/dsd/cohort/application/bootstrap/BootstrapData.java:1]: Did you want to leave the console print outs here?

Comment (Dissurender) [reply Backend_Application/src/main/java/dsd/cohort/application/bootstrap/BootstrapData.java:1]: I have server side prints to monitor activity on when the logic to fetch triggers. No secure data shown

**Review by wcouturiaux — DISMISSED** [review]
Breaker Breaker 1-9 this here's the rubber 🦆

**Review by bethanyann — APPROVED** [review]

**Review by wcouturiaux — APPROVED** [review]

**Pull request merged**